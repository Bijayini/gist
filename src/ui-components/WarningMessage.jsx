import React, { Fragment } from 'react';
import PropTypes from 'prop-types';

import theme from '../lib/theme';
import BodyCopy from './BodyCopy';

const WarningMessage = ({ text }) => {
  return (
    <Fragment>
      <div className="warning-message">
        <BodyCopy
          xs="3"
          xm="3"
          xl="3"
          color="warningMessageColor"
          variant="italic"
        >
          {text}
        </BodyCopy>
      </div>
      <style jsx="true">
        {`
          .warning-message {
            background: ${theme.warningMessageBg};
            padding: 10px;
            text-align: center;
          }
        `}
      </style>
    </Fragment>
  );
};

WarningMessage.propTypes = {
  text: PropTypes.string.isRequired,
};

export default WarningMessage;
