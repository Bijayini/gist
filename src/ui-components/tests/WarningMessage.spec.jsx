import React from 'react';
import { shallow } from 'enzyme';

import WarningMessage from '../WarningMessage';

describe('<WarningMessage />', () => {
  it('should match with snapshot', () => {
    const text = 'random text';

    const warningComponent = shallow(<WarningMessage text={text} />)

    expect(warningComponent.debug()).toMatchSnapshot();
  });
});
