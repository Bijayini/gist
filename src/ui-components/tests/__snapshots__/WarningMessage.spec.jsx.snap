// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarningMessage /> should match with snapshot 1`] = `
"<Fragment>
  <div className=\\"jsx-237712953 warning-message\\">
    <BodyCopy xs=\\"3\\" xm=\\"3\\" xl=\\"3\\" color=\\"warningMessageColor\\" variant=\\"italic\\" className=\\"\\" uppercase={false} styles={false}>
      random text
    </BodyCopy>
  </div>
  <JSXStyle id=\\"3136427526\\" dynamic={{...}}>
    .warning-message.__jsx-style-dynamic-selector{background:#fbdae1;padding:10px;text-align:center;}
  </JSXStyle>
</Fragment>"
`;
